x/gamm/twap/types/utils_test.go

```diff
@@ -0,0 +1,34 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v10/osmoutils"
+)
+
+func TestGetAllUniqueDenomPairs(t *testing.T) {
+	tests := map[string]struct {
+		denoms       []string
+		wantedPairGT []string
+		wantedPairLT []string
+		panics       bool
+	}{
+		"basic":    {[]string{"A", "B"}, []string{"B"}, []string{"A"}, false},
+		"basicRev": {[]string{"B", "A"}, []string{"B"}, []string{"A"}, false},
+		// AB > A
+		"prefixed": {[]string{"A", "AB"}, []string{"AB"}, []string{"A"}, false},
+		"basic-3":  {[]string{"A", "B", "C"}, []string{"C", "C", "B"}, []string{"B", "A", "A"}, false},
```

Comment:
I don't understand the basic-3 test case. How is the the resulting GT pair and LT Pair the correct unique denom pairs

Reply:
Sorts by gt symbol first then lt symbol

So
C,b
C,a
B,a